src/Core/Material/MaterialModel.hpp

```diff
@@ -30,6 +35,15 @@ class RA_CORE_API MaterialModel : public Utils::ObservableVoid
     /// DEBUG
     virtual void displayInfo() const;
 
+    /** Compute BSDF value for a set of directions and texture coordinates.
+     *  w_i : incident direction, w_o : outgoing direction, normal : geometric normal direction.
+     *  w_i, w_o and normal are in world space.
+     */
+    virtual Utils::Color evalBSDF( Vector3 w_i, Vector3 w_o, Vector3 normal, Vector2 uv ) = 0;
```

Comment:
either
```suggestion
    virtual Utils::Color evaluate( Vector3 w_i, Vector3 w_o, Vector3 normal, Vector2 uv ) = 0;
```
or (I prefer this but can live with a named method ^^)
```suggestion
    virtual Utils::Color operator()( Vector3 w_i, Vector3 w_o, Vector3 normal, Vector2 uv ) = 0;
```